src/index/ivf/ivf.cc

```diff
@@ -100,9 +100,7 @@ class IvfIndexNode : public IndexNode {
             auto nlist = index_->nlist;
             auto code_size = index_->code_size;
             return (nb * code_size + nb * sizeof(int64_t) + nlist * code_size);
-        }
-
-        if constexpr (std::is_same<T, faiss::IndexIVFPQ>::value) {
+        } else if constexpr (std::is_same<T, faiss::IndexIVFPQ>::value) {
```

Comment:
not need else if constexpr, this if condition selected after code compiled.
line 45 static_assert is limited the template type.

Reply:
ok, roll it back